fl4health/utils/nnunet_utils.py

```diff
@@ -461,12 +461,13 @@ def get_lr(self) -> Sequence[float]:
                 f"Current LR step of {self._step_count} reached Max Steps of {self.max_steps}. LR will remain fixed.",
             )
 
+        # Subtract 1 from step count since it starts at 1
```

Comment:
I might be missing something, but are we actually subtracting one here?

Reply:
Oops lol ... updated and added assert that the log message doesn't trigger early and only after the number of steps exceeds max steps

Reply:
Seconded. Also can you add a comment noting that pytorch imposes that step_count starts at 1.

Reply:
Also it seems like step count might actually start at 2 since pytorch calls step on initialization of the LR scheduler. Not sure why they do this.

Reply:
Maybe I'm missing somehing, but it seems like the self._initialize_step method of the pytorch base class performs a step on init before training. So by the time step is called after the first training step, self._step_count will equal 2

Reply:
If true...that's really silly 😂

Reply:
mmm that makes sense. Perhaps their counter starts at zero and bumps from there. Still a weird design choice.

Reply:
They start it at zero, and call step on init, but within the step function, self._step_count is incremented before calling self.get_lr. That being said maybe there is something else going on under the hood that I'm missing. I trust that your test confirms that we have the correct LR at each step.

Reply:
@scarere what a nightmare ... let me know if it pops up on your side, its not exhaustive testing but still would be surprised if a magic +1 gets added in under certain conditions